### PR TITLE
add additonal condition for prev msdd cfgs

### PR DESCRIPTION
**Pull request by nithinraok**
# What does this PR do ?

TitaNet model was updated but some previous configurations were not. 
This PR provides support for prevevious msdd configuration versions.  

**Collection**: SD

# Changelog 
- Catch the `_target_` code in newer titanet versions, which is not available in previous versions. 

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

### CODE REVIEW

**Comment by tango4j** [comment]
- Changed Speaker embedding extraction to avoid `torch.cat` in every for loop step for speed up. 
This does not make a lot of difference in general, but if CPU is busy with other tasks, this could make some difference in processing time.

- Now, device=null and if torch.cuda.is_cuda_available() == True, then MSDD is running on GPU. This means diarization default setting will now run on GPU if cuda is available.

**Review by tango4j — DISMISSED** [review]

**Review by titu1994 — DISMISSED** [review]
